ARCHITECTURE.md

```diff
@@ -38,6 +38,11 @@ Character Streams -> Lexer -> (Token Stream) -> Parser -> (AST) -> Semantic Pass
 	+ May reduce some higher level Ast concepts into easier concepts to express in the IR.
 	+ *** Need to cover visitors, and require, scheduler, etc.
 	+ Cover how visitors work -- Equivalent to functional programming concept of parameter specialization
+	+ Operates in multiple passes over the AST/IR (Ew that it modifies the IR/AST?)
```

Comment:
The AST and the IR are separate.
Rather the AST is the basis for the IR.
The AST is not touched after parsing.
The IR is indeed subject to lowering and other transformations